creator-node/src/routes/tracks.js

```diff
@@ -550,26 +551,99 @@ module.exports = function (app) {
       return sendResponse(req, res, errorResponseBadRequest(`Invalid ID: ${encodedId}`))
     }
 
-    const { blockchainId: blockchainIdFromTrack } = await models.Track.findOne({
-      attributes: ['blockchainId'],
-      where: { blockchainId },
-      order: [['clock', 'DESC']]
-    })
-
-    if (!blockchainIdFromTrack) {
-      return sendResponse(req, res, errorResponseBadRequest(`No track found for blockchainId ${blockchainId}`))
-    }
-
-    const { multihash } = await models.File.findOne({
+    let fileRecord = await models.File.findOne({
       attributes: ['multihash'],
       where: {
         type: 'copy320',
-        trackBlockchainId: blockchainIdFromTrack
+        trackBlockchainId: blockchainId
       },
       order: [['clock', 'DESC']]
     })
 
-    if (!multihash) {
+    try {
+      // see if there's a fileRecord in redis so we can short circuit all this logic
+      let redisFileRecord = await redisClient.get(`streamFallback:::${blockchainId}`)
+      if (redisFileRecord) {
+        redisFileRecord = JSON.parse(redisFileRecord)
+        if (redisFileRecord && redisFileRecord.multihash) {
+          fileRecord = redisFileRecord
+        }
+      }
+    } catch (e) {
+      req.logger.error(`Error looking for stream fallback in redis`, e)
+    }
+
+    // if track didn't finish the upload process and was never associated, there may not be a trackBlockchainId for the File records,
+    // try to fall back to discovery to fetch the metadata multihash and see if you can deduce the copy320 file
+    if (!fileRecord) {
+      try {
+        let trackRecord = await libs.Track.getTracks(1, 0, [blockchainId])
+        if (!trackRecord || trackRecord.length === 0 || !trackRecord[0].hasOwnProperty('blocknumber')) {
+          // TODO - return a server err response
+          return sendResponse(req, res, errorResponseServerError('Missing or malformatted track fetched from discovery node.'))
+        }
+
+        trackRecord = trackRecord[0]
+
+        // query the files table for the metadata multihash from discovery
+        // no CNodeUserUUID, but because this track is associated with a user and contains that user_id
+        // inside it, it's unique to this user
+        const file = await models.File.findOne({ where: { multihash: trackRecord.metadata_multihash } })
+        if (!file) {
+          return sendResponse(req, res, errorResponseServerError('Missing or malformatted track fetched from discovery node.'))
+        }
+
+        // read the metadata multihash from from the file system
+        let metadataJSON
+
+        const fileBuffer = await readFile(file.storagePath)
```

Comment:
probably can get away without doing this file read + validation ?

Reply:
to clarify, this is bc the discprov response already contains all segments in array so this retrieval + validation seems unnecessary

Reply:
not sure what this means

Reply:
basically - why do we need to retrieve the metadata file from disk? the discprov response already gives us all the segment CIDs, so whats the purpose of retrieving metadata file record from db or disk?

Reply:
amazing